receiver/dockerstatsreceiver/receiver.go

```diff
@@ -115,8 +120,260 @@ func (r *receiver) scrape(ctx context.Context) (pmetric.Metrics, error) {
 			errs = multierr.Append(errs, scrapererror.NewPartialScrapeError(res.err, 0))
 			continue
 		}
-		res.md.ResourceMetrics().CopyTo(md.ResourceMetrics())
+		res.md.ResourceMetrics().MoveAndAppendTo(md.ResourceMetrics())
 	}
 
 	return md, errs
 }
+
```

Comment:
This is more a stylistic choice but I would create a seperate file for the `resultv2` and associated changes since if the intention is to deprecate and remove that old functionality, it should be a clean and easy removal from git instead of line changes and import statements.

Reply:
Yeah, I like that idea. I will do that.

Reply:
Done